admin_bans.php

```diff
@@ -76,6 +76,10 @@
 				$result = $db->query('SELECT registration_ip FROM '.$db->prefix.'users WHERE id='.$user_id) or error('Unable to fetch user info', __FILE__, __LINE__, $db->error());
 				$ban_ip = ($db->num_rows($result)) ? $db->result($result) : '';
 			}
+
+			// Disable localhost IP ban
+			if (in_array($ban_ip, array('', '127.0.0.1', '::1'), true))
+				$ban_ip = '';

```

Comment:
I would use `null` rather than an empty string. That way the check if `''` is in the array also makes more sense :smile:  (this way you're checking if `''` is the `$ban_ip` and if it is, you're setting `$ban_ip` to be an empty string once again, that's pretty duplicate.